module/VuFind/src/VuFind/ILS/Driver/Demo.php

```diff
@@ -1674,6 +1683,56 @@ public function getCancelHoldDetails($hold, $patron = [])
             ? $hold['reqnum'] : '';
     }
 
+    /**
+     * Update holds
+     *
+     * This is responsible for changing the status of hold requests
+     *
+     * @param array $holdsDetails The details identifying the holds
+     * @param array $fields       An associative array of fields to be updated
+     * @param array $patron       Patron array
+     *
+     * @return array Associative array of the results
+     */
+    public function updateHolds(array $holdsDetails, array $fields, array $patron
+    ): array {
+        $results = [];
+        $session = $this->getSession($patron['id']);
+        foreach ($session->holds as &$currentHold) {
+            if (!isset($currentHold['updateDetails'])
+                || !in_array($currentHold['updateDetails'], $holdsDetails)
+            ) {
+                continue;
+            }
+            if ($this->isFailing(__METHOD__, 25)) {
```

Comment:
Since we have a new method that can fail, we should add this method to the Failure_Probabilities section of Demo.ini for consistency.

Reply:
Done.